### PR TITLE
Multiple component update, Creating a new component

### PR DESCRIPTION
**Pull request by Duncid**
- Updating Chip with a busy state
- Updating Button with a Button "List"
- Creating a new ContextItem

### CODE REVIEW

File: sparkle/src/stories/ContextItem.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  TrashIcon,
+} from "../index_with_tw_base";
+
+const meta = {
+  title: "Molecule/ContextItem",
+  component: ContextItem,
+} satisfies Meta<typeof ContextItem>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
```
Comment (spolu) [inline sparkle/src/stories/ContextItem.stories.tsx:24]: Story seems unused

File: sparkle/src/components/Button.tsx [hunk 2]
```diff
@@ -186,3 +186,16 @@ export function Button({
     </Tooltip>
   );
 }
+
+interface ButtonBarProps {
+  children: ReactNode;
+  className?: string;
+}
+
+Button.List = function ({ children, className }: ButtonBarProps) {
```
Comment (fontanierh) [inline sparkle/src/components/Button.tsx:195]: I'm not sure I understand what this is -- is it essentially just a list of buttons ?

**Review by spolu — DISMISSED** [review]
LGTM but please clean the unused import :pray:

File: sparkle/src/components/ContextItem.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+interface ContextItemListProps {
+  children: ReactNode;
+  className?: string;
+}
+
+ContextItem.List = function ({ children, className }: ContextItemListProps) {
+  // Ensure all children are of type ContextItem
+  React.Children.forEach(children, (child) => {
```
Comment (fontanierh) [inline sparkle/src/components/ContextItem.tsx:51]: I'm a bit worried that this isn't enforce by typescript. I don't think we really typecheck the children. Is there a strong reason not to accept an array of props and have the `ContextItem` be created from the component instead ?

File: sparkle/src/components/Button.tsx (hunk 2 above)
Comment (Duncid) [reply sparkle/src/components/Button.tsx:195]: Exactly

File: sparkle/src/components/ContextItem.tsx (hunk 3 above)
Comment (Duncid) [reply sparkle/src/components/ContextItem.tsx:51]: It activate or deactivate "hasSeparator' of the item to create a list with separator.
So it does need to make sure it's a ContextItem (to know it has a hasSeparator.

**Pull request merged**